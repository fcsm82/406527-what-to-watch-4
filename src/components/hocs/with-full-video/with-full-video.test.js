import React from "react";
import renderer from "react-test-renderer";
import withFullVideo from "./with-full-video.js";
import PropTypes from "prop-types";

const MockComponent = (props) => {
  const {children} = props;
  return (
    <>
        {children}
    </>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withFullVideo(MockComponent);

it(`WithFullVideo is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      preview = {``}
      title = {``}
      onExitClick = {() => {}}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
