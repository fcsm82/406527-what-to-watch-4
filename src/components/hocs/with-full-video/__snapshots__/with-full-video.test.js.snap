// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithFullVideo is rendered correctly 1`] = `
<video
  className="player__video"
  poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
/>
`;
